src/utils/fetchUtils.ts

```diff
@@ -88,6 +89,7 @@ export class FetchResponse extends Response {
       }
     }
 
-    FetchResponse.setUrl(init.url, this)
+    FetchResponse.setUrl(init.url, response)
+    return response
```

Comment:
I wonder if it wouldn't be better for us to flip the solution around:

- Export the header recording function to be able to apply it to any instance of `Request`/`Response`. 

So all we have to do in this class is 

```ts
return recordRawHeaders(this)
```

@mikicho, what do you think about this approach?

Reply:
I like this approach more.. it's straightforward (kiss <3).